persist/sqlite/consensus_test.go

```diff
@@ -882,12 +880,80 @@ func BenchmarkRevert(b *testing.B) {
 		scID = txn3.SiacoinOutputID(txn3.ID(), 0)
 		val = txn3.SiacoinOutputs[0].Value
 
-		block := testutil.MineV2Block(n.tipState(), []types.Transaction{txn1, txn2}, []types.V2Transaction{txn3}, types.VoidAddress)
-		n.applyBlock(b, block)
+		v2FC, payout := prepareV2Contract(pk1, pk1, n.tipState().Index.Height+1)
+		txn4 := types.V2Transaction{
+			SiacoinInputs: []types.V2SiacoinInput{{
+				Parent: types.SiacoinElement{
+					ID: txn3.SiacoinOutputID(txn3.ID(), 0),
+					StateElement: types.StateElement{
+						LeafIndex: types.UnassignedLeafIndex,
+					},
+					SiacoinOutput: txn3.SiacoinOutputs[0],
+				},
+				SatisfiedPolicy: types.SatisfiedPolicy{Policy: addr1Policy},
+			}},
+			SiacoinOutputs: []types.SiacoinOutput{{
+				Address: addr1,
+				Value:   val.Sub(payout),
+			}},
+			FileContracts: []types.V2FileContract{v2FC},
+		}
+		testutil.SignV2TransactionWithContracts(n.tipState(), pk1, pk1, pk1, &txn4)
+
+		scID = txn4.SiacoinOutputID(txn4.ID(), 0)
+		val = txn4.SiacoinOutputs[0].Value
+
+		return testutil.MineV2Block(n.tipState(), []types.Transaction{txn1, txn2}, []types.V2Transaction{txn3, txn4}, types.VoidAddress)
 	}
 
-	block := n.blocks[len(n.blocks)-1]
-	bs := n.supplements[len(n.supplements)-1]
+	for i := range nBlocks {
+		if i%100 == 0 {
+			b.Logf("Mined %d blocks", i)
+		}
+
+		n.applyBlock(b, generateBlock())
+	}
+
+	block := generateBlock()
+	bs := n.store.SupplementTipBlock(block)
+	cs := n.states[len(n.states)-1]
+
+	if err := consensus.ValidateBlock(cs, block, bs); err != nil {
+		b.Fatal(err)
+	}
+	cs, au := consensus.ApplyBlock(cs, block, bs, time.Time{})
+	caus := []chain.ApplyUpdate{{
+		ApplyUpdate: au,
+		Block:       block,
+		State:       cs,
+	}}
+
+	b.Run("apply", func(b *testing.B) {
+		b.ResetTimer()
+		for range b.N {
+			err := n.db.transaction(func(tx *txn) error {
+				defer tx.Rollback()
+				utx := &updateTx{
+					tx: tx,
+				}
+
+				b.StartTimer()
```

Comment:
Looks slightly weird to me because it's usually the inverse, but it's probably fine though. I would replace `b.ResetTimer` with `b.StopTimer`, because resetting is redundant and if you stop it timer it's even more correct.